src/modules/privacy-idea-administration/privacy-idea-administration.controller.ts

```diff
@@ -173,8 +175,19 @@ export class PrivacyIdeaAdministrationController {
         @Permissions() permissions: PersonPermissions,
     ): Promise<void> {
         const referrer: string = await this.getReferrerIfAllowedOrSelf(params.personId, permissions);
-
-        await this.privacyIdeaAdministrationService.verifyTokenEnrollment(referrer, params.otp);
+        try {
+            await this.privacyIdeaAdministrationService.verifyTokenEnrollment(referrer, params.otp);
+            this.logger.info(
+                `Benutzer ${referrer} (BenutzerId: ${params.personId}) hat sich einen Software-Token für 2FA eingerichtet.`,
+            );
+        } catch (error) {
+            if (error instanceof Error) {
+                this.logger.error(
+                    `Benutzer mit BenutzerID: ${params.personId} hat versucht eine 2FA einzurichten. Fehler: ${error.message}`,
+                );
+            }
+            throw error;
```

Comment:
Similar to the problem above, thrown errors that aren't instances of `Error` (will probably never happen, but better to be sure) won't be logged.

Reply:
adjusted